includes/discovery/functions.inc.php

```diff
@@ -1482,3 +1482,60 @@ function find_port_id($description, $identifier = '', $device_id = 0, $mac_addre
 
     return (int)dbFetchCell($sql, $params);
 }
+
+/**
+ * Try to find a port by lldp port id
+ *
+ * @param string $lldpremportid matched against ports.lldpLocPortId
+ * @param int $device_id restrict search to ports on a specific device
+ * @return int
+ */
+function discover_lldp_loc_port_id($device)
+{
+
+    $port_stats = [];
+    $port_stats = snmpwalk_cache_oid($device, 'lldpLocPortId', $port_stats, 'LLDP-MIB');
+
+    foreach ($port_stats as $ifIndex => $snmp_data) {
+        $port = dbFetchRow('SELECT * from `ports` where `device_id` = ? AND `ifIndex` = ?', [$device['device_id'], $ifIndex]);
+        if ($port['port_id']) {
```

Comment:
I don't see an "else" statement for this condition. So if the port does not exist yet in the DB, you do not store the lldp data currently.

Reply:
Yeah, that was my plan, but after thinking about some other cases, I think I am going to rework that a bit.
Right now it will work if a switch module gets plugged in and adds ports to a switch, but if one is taken out, those old ports will still be there. I am going to look into how the current ports mod handles that and make some changes.

Reply:
Correct, that's the point that would have come just after. Basically, every time you do LLDP discovery, you maintain an array of the entries that are in the DB and in the LLDP. And you delete any entry that is in the DB and not match in the snmp, and add/update any from the SNMP to the DB.